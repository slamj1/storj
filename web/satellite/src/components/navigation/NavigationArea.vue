// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template src="./navigationArea.html"></template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import ProjectSelectionArea from '@/components/header/projectSelection/ProjectSelectionArea.vue';

import DocsIcon from '@/../static/images/navigation/docs.svg';
import LogoIcon from '@/../static/images/navigation/logo.svg';
import LogoTextIcon from '@/../static/images/navigation/logoText.svg';
import SupportIcon from '@/../static/images/navigation/support.svg';

import { RouteConfig } from '@/router';
import { NavigationLink } from '@/types/navigation';

@Component({
    components: {
        ProjectSelectionArea,
        LogoIcon,
        LogoTextIcon,
        DocsIcon,
        SupportIcon,
    },
})
export default class NavigationArea extends Vue {
    /**
     * Indicates if resource related navigation links appears.
     */
    public areResourceItemsShown: boolean = true;
    public isResourceButtonShown: boolean = false;

    /**
     * Indicates if account related navigation links appears.
     */
    public areAccountItemsShown: boolean = true;
    public isAccountButtonShown: boolean = false;

    /**
     * Toggles resource items visibility.
     */
    public toggleResourceItemsVisibility(): void {
        this.areResourceItemsShown = !this.areResourceItemsShown;
    }

    /**
     * Toggles resource button visibility.
     */
    public toggleResourceButtonVisibility(): void {
        this.isResourceButtonShown = !this.isResourceButtonShown;
    }

    /**
     * Toggles account items visibility.
     */
    public toggleAccountItemsVisibility(): void {
        this.areAccountItemsShown = !this.areAccountItemsShown;
    }

    /**
     * Toggles account button visibility.
     */
    public toggleAccountButtonVisibility(): void {
        this.isAccountButtonShown = !this.isAccountButtonShown;
    }

    /**
     * Array of navigation links with icons.
     */
    // TODO: Use SvgLoaderComponent to reduce markup lines
    public readonly navigation: NavigationLink[] = [
        RouteConfig.ProjectDashboard.withIcon(
            `<svg class="svg" width="19" height="17" viewBox="0 0 19 17" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path class="navigation-svg-path" fill-rule="evenodd" clip-rule="evenodd" d="M15.6691 16.8889H9.5H3.3309C1.29229 15.1294 0 12.5137 0 9.59331C0 4.29507 4.2533 0 9.5 0C14.7467 0 19 4.29507 19 9.59331C19 12.5137 17.7077 15.1294 15.6691 16.8889ZM12.9227 5.32961L14.4156 6.83704L9.93721 11.3594L8.44444 9.85192L12.9227 5.32961Z" fill="#7C8794"/>
              </svg>
             `),
        RouteConfig.Team.withIcon(
            `<svg class="svg" width="21" height="18" viewBox="0 0 21 18" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path class="navigation-svg-path" opacity="0.8" fill-rule="evenodd" clip-rule="evenodd" d="M14.8165 0C17.7839 0 20.1895 2.40557 20.1895 5.373C20.1895 7.40389 19.0627 9.17161 17.4002 10.0851C19.4392 11.0968 20.8407 13.2 20.8407 15.6305V17.91H8.46655V15.6305C8.46655 13.13 9.94989 10.976 12.0846 10.0004C10.5035 9.06519 9.44345 7.34289 9.44345 5.373C9.44345 2.40557 11.849 0 14.8165 0ZM6.34991 0C7.36246 0 8.3096 0.280088 9.11806 0.767013C8.09966 2.02549 7.48964 3.62801 7.48964 5.373L7.49091 5.51065C7.51986 7.06803 8.03908 8.53529 8.92906 9.73552L8.97519 9.79612L8.94777 9.82267C7.42471 11.3194 6.51273 13.3922 6.51273 15.6305V17.91H0V15.6305C0 13.13 1.48335 10.976 3.61809 10.0004C2.037 9.06519 0.976909 7.34289 0.976909 5.373C0.976909 2.40557 3.38248 0 6.34991 0Z" fill="#909BA8"/>
              </svg>
             `),
        RouteConfig.ApiKeys.withIcon(
            `<svg class="svg" width="19" height="19" viewBox="0 0 19 19" fill="none" xmlns="http://www.w3.org/2000/svg">
                <path class="navigation-svg-path" opacity="0.8" fill-rule="evenodd" clip-rule="evenodd" d="M19 0L18.8777 3.30167L17.3804 3.33154L17.3114 4.86798L15.548 5.164L15.5552 6.62419L14.0737 6.63827L13.8958 8.28354L12.3349 9.84422L13.2397 13.2214C13.3996 13.8182 13.229 14.4549 12.7921 14.8918L9.19044 18.4934C8.75356 18.9303 8.1168 19.1009 7.52001 18.941L2.60009 17.6227C2.00331 17.4628 1.53716 16.9967 1.37725 16.3999L0.0589658 11.48C-0.100943 10.8832 0.0696777 10.2464 0.506556 9.80956L4.10819 6.20793C4.54507 5.77105 5.18183 5.60043 5.77862 5.76034L9.15552 6.66484L15.6983 0.122284L19 0ZM6.52703 12.473C5.78414 11.7301 4.57967 11.7301 3.83678 12.473C3.09389 13.2159 3.09389 14.4203 3.83678 15.1632C4.57967 15.9061 5.78414 15.9061 6.52703 15.1632C7.26992 14.4203 7.26992 13.2159 6.52703 12.473Z" fill="#909BA8"/>
              </svg>
             `),
    ];

    /**
     * Array of account related navigation links.
     */
    public readonly accountNavigation: NavigationLink[] = [
        RouteConfig.Account.with(RouteConfig.Settings),
        RouteConfig.Account.with(RouteConfig.Billing),
        // TODO: disabled until implementation
        // RouteConfig.Account.with(RouteConfig.Referral),
    ];

    /**
     * Returns home path depending on app's state.
     */
    public get homePath(): string {
        if (this.isOnboardingTour) {
            return RouteConfig.OnboardingTour.path;
        }

        return RouteConfig.ProjectDashboard.path;
    }

    /**
     * Indicates if resources displaying button is shown.
     */
    public get isResourcesDisplayingButtonShown(): boolean {
        return !this.areResourceItemsShown && this.isResourceButtonShown;
    }

    /**
     * Indicates if resources items hiding button is shown.
     */
    public get isResourcesHidingButtonShown(): boolean {
        return this.areResourceItemsShown && this.isResourceButtonShown;
    }

    /**
     * Indicates if account items displaying button is shown.
     */
    public get isAccountItemsDisplayingButtonShown(): boolean {
        return !this.areAccountItemsShown && this.isAccountButtonShown;
    }

    /**
     * Indicates if account items hiding button is shown.
     */
    public get isAccountItemsHidingButtonShown(): boolean {
        return this.areAccountItemsShown && this.isAccountButtonShown;
    }

    /**
     * Indicates if roter link is disabled.
     */
    public get isLinkDisabled(): boolean {
        return this.isOnboardingTour || this.isNoProject;
    }

    /**
     * Indicates if current route is onboarding tour.
     */
    private get isOnboardingTour(): boolean {
        return this.$route.name === RouteConfig.OnboardingTour.name;
    }

    /**
     * Indicates if there is no projects.
     */
    private get isNoProject(): boolean {
        return this.$store.state.projectsModule.projects.length === 0;
    }
}
</script>

<style src="./navigationArea.scss" lang="scss"></style>
