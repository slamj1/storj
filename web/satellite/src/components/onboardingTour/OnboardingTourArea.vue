// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="tour-area">
        <ProgressBar
            :is-create-project-step="isCreateProjectState"
            :is-create-api-key-step="isCreatApiKeyState"
            :is-upload-data-step="isUploadDataState"
        />
        <OverviewStep
            v-if="isDefaultState"
            @setAddPaymentState="setAddPaymentState"
        />
        <AddPaymentStep
            v-if="isAddPaymentState"
            @setProjectState="setCreateProjectState"
        />
        <CreateProjectStep
            v-if="isCreateProjectState"
            @setApiKeyState="setCreateApiKeyState"
        />
        <CreateApiKeyStep
            v-if="isCreatApiKeyState"
            @setUploadDataState="setUploadDataState"
        />
        <UploadDataStep v-if="isUploadDataState"/>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import ProgressBar from '@/components/onboardingTour/ProgressBar.vue';
import AddPaymentStep from '@/components/onboardingTour/steps/AddPaymentStep.vue';
import CreateApiKeyStep from '@/components/onboardingTour/steps/CreateApiKeyStep.vue';
import CreateProjectStep from '@/components/onboardingTour/steps/CreateProjectStep.vue';
import OverviewStep from '@/components/onboardingTour/steps/OverviewStep.vue';
import UploadDataStep from '@/components/onboardingTour/steps/UploadDataStep.vue';

import CheckedImage from '@/../static/images/common/checked.svg';

import { RouteConfig } from '@/router';
import { TourState } from '@/utils/constants/onboardingTourEnums';

@Component({
    components: {
        UploadDataStep,
        CreateApiKeyStep,
        CreateProjectStep,
        AddPaymentStep,
        ProgressBar,
        OverviewStep,
        CheckedImage,
    },
})

export default class OnboardingTourArea extends Vue {
    public areaState: number = TourState.DEFAULT;

    /**
     * Lifecycle hook after initial render.
     * Sets area to needed state.
     */
    public mounted(): void {
        if (this.userHasProject && this.userHasApiKeys) {
            try {
                this.$router.push(RouteConfig.ProjectDashboard.path);
            } catch (error) {
                return;
            }

            return;
        }

        if (this.userHasProject && !this.userHasApiKeys) {
            this.setCreateApiKeyState();

            return;
        }

        if (this.$store.state.paymentsModule.creditCards.length > 0) {
            this.setCreateProjectState();

            return;
        }

        if (this.$store.getters.isTransactionProcessing || this.$store.getters.isTransactionCompleted) {
            this.setAddPaymentState();
        }
    }

    /**
     * Indicates if area is in default state.
     */
    public get isDefaultState(): boolean {
        return this.areaState === TourState.DEFAULT;
    }

    /**
     * Indicates if area is in adding payment method state.
     */
    public get isAddPaymentState(): boolean {
        return this.areaState === TourState.ADDING_PAYMENT;
    }

    /**
     * Indicates if area is in creating project state.
     */
    public get isCreateProjectState(): boolean {
        return this.areaState === TourState.PROJECT;
    }

    /**
     * Indicates if area is in api key state.
     */
    public get isCreatApiKeyState(): boolean {
        return this.areaState === TourState.API_KEY;
    }

    /**
     * Indicates if area is in upload data state.
     */
    public get isUploadDataState(): boolean {
        return this.areaState === TourState.UPLOAD;
    }

    /**
     * Sets area's state to adding payment method state.
     */
    public setAddPaymentState(): void {
        this.areaState = TourState.ADDING_PAYMENT;
    }

    /**
     * Sets area's state to creating project state.
     */
    public setCreateProjectState(): void {
        this.areaState = TourState.PROJECT;
    }

    /**
     * Sets area's state to creating api key state.
     */
    public setCreateApiKeyState(): void {
        this.areaState = TourState.API_KEY;
    }

    /**
     * Sets area's state to upload data state.
     */
    public setUploadDataState(): void {
        this.areaState = TourState.UPLOAD;
    }

    /**
     * Indicates if user has at least one project.
     */
    private get userHasProject(): boolean {
        return this.$store.state.projectsModule.projects.length > 0;
    }

    /**
     * Indicates if user has at least one API key.
     */
    private get userHasApiKeys(): boolean {
        return this.$store.state.apiKeysModule.page.apiKeys.length > 0;
    }
}
</script>

<style scoped lang="scss">
    .tour-area {
        padding: 0 100px;
    }

    @media screen and (max-width: 1380px) {

        .tour-area {
            padding: 0 50px;
        }
    }

    @media screen and (max-width: 1000px) {

        .tour-area {
            padding: 0 25px;
        }
    }
</style>
