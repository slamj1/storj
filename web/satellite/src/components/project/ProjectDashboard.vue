// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="dashboard-area">
        <h1 class="dashboard-area__title">Dashboard</h1>
        <div class="dashboard-area__top-area">
            <ProjectDetails/>
            <ProjectUsage/>
        </div>
        <BucketArea/>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import BucketArea from '@/components/project/buckets/BucketArea.vue';
import ProjectDetails from '@/components/project/ProjectDetails.vue';
import ProjectUsage from '@/components/project/usage/ProjectUsage.vue';

import { RouteConfig } from '@/router';
import { SegmentEvent } from '@/utils/constants/analyticsEventNames';

@Component({
    components: {
        BucketArea,
        ProjectDetails,
        ProjectUsage,
    },
})
export default class ProjectDashboard extends Vue {
    /**
     * Lifecycle hook after initial render.
     * Segment tracking is processed.
     */
    public mounted(): void {
        if (!this.$store.getters.selectedProject.id) {
            this.$router.push(RouteConfig.OnboardingTour.path);

            return;
        }

        this.$segment.track(SegmentEvent.PROJECT_VIEWED, {
            project_id: this.$store.getters.selectedProject.id,
        });
    }
}
</script>

<style scoped lang="scss">
    .dashboard-area {
        padding: 40px 65px 80px 65px;

        &__title {
            font-family: 'font_bold', sans-serif;
            font-size: 32px;
            line-height: 39px;
            color: #263549;
            margin: 0 0 35px 0;
        }

        &__top-area {
            display: flex;
            align-items: flex-start;
            justify-content: space-between;
        }
    }
</style>
