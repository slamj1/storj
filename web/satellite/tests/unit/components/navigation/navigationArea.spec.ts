// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import Router from 'vue-router';
import Vuex from 'vuex';

import NavigationArea from '@/components/navigation/NavigationArea.vue';
import OnboardingTourArea from '@/components/onboardingTour/OnboardingTourArea.vue';
import ProjectDashboard from '@/components/project/ProjectDashboard.vue';

import { RouteConfig } from '@/router';
import { makeProjectsModule, PROJECTS_MUTATIONS } from '@/store/modules/projects';
import { NavigationLink } from '@/types/navigation';
import { Project } from '@/types/projects';
import { createLocalVue, shallowMount } from '@vue/test-utils';

import { ProjectsApiMock } from '../../mock/api/projects';

const api = new ProjectsApiMock();
api.setMockProjects([new Project('1')]);
const projectsModule = makeProjectsModule(api);
const localVue = createLocalVue();

localVue.use(Vuex);
localVue.use(Router);

const store = new Vuex.Store({ modules: { projectsModule } });

const expectedLinks: NavigationLink[] = [
    RouteConfig.ProjectDashboard,
    RouteConfig.Team,
    RouteConfig.ApiKeys,
];

describe('NavigationArea', () => {
    it('snapshot not changed without project', (): void => {
        const router = new Router({
            mode: 'history',
            routes: [{
                path: '/',
                name: RouteConfig.OnboardingTour.name,
                component: OnboardingTourArea,
            }],
        });
        const wrapper = shallowMount(NavigationArea, {
            store,
            localVue,
            router,
        });

        const navigationElements = wrapper.findAll('.navigation-area__item-container');
        const disabledElements = wrapper.findAll('.navigation-area__item-container.disabled');
        const resourcesButton = wrapper.findAll('.navigation-area__resources-title__button');
        const accountButton = wrapper.findAll('.navigation-area__account-title__button');

        expect(navigationElements.length).toBe(7);
        expect(disabledElements.length).toBe(7);
        expect(resourcesButton.length).toBe(0);
        expect(accountButton.length).toBe(0);

        expect(wrapper).toMatchSnapshot();
    });

    const router = new Router({
        mode: 'history',
        routes: [{
            path: '/',
            name: RouteConfig.ProjectDashboard.name,
            component: ProjectDashboard,
        }],
    });

    it('snapshot not changed with project', async () => {
        const projects = await store.dispatch('fetchProjects');
        store.commit(PROJECTS_MUTATIONS.SELECT_PROJECT, projects[0].id);

        const wrapper = shallowMount(NavigationArea, {
            store,
            localVue,
            router,
        });

        const navigationElements = wrapper.findAll('.navigation-area__item-container');
        const disabledElements = wrapper.findAll('.navigation-area__item-container.disabled');
        const resourcesButton = wrapper.findAll('.navigation-area__resources-title__button');
        const accountButton = wrapper.findAll('.navigation-area__account-title__button');

        expect(navigationElements.length).toBe(7);
        expect(disabledElements.length).toBe(0);
        expect(resourcesButton.length).toBe(0);
        expect(accountButton.length).toBe(0);

        expect(wrapper).toMatchSnapshot();
    });

    it('navigation links are correct', () => {
        const wrapper = shallowMount(NavigationArea, {
            store,
            localVue,
            router,
        });

        const navigationLinks = (wrapper.vm as any).navigation;

        expect(navigationLinks.length).toBe(expectedLinks.length);

        expectedLinks.forEach((link, i) => {
            expect(navigationLinks[i].name).toBe(expectedLinks[i].name);
            expect(navigationLinks[i].path).toBe(expectedLinks[i].path);
        });
    });

    it('trigger show/hide events works correctly', async () => {
        const wrapper = shallowMount(NavigationArea, {
            store,
            localVue,
            router,
        });

        await wrapper.find('.navigation-area__resources-title').trigger('mouseenter');
        await wrapper.find('.navigation-area__account-title').trigger('mouseenter');

        expect(wrapper.find('.navigation-area__resources-title__button').text()).toMatch('Hide');
        await wrapper.find('.navigation-area__resources-title__button').trigger('click');

        expect(wrapper.find('.navigation-area__account-title__button').text()).toMatch('Hide');
        await wrapper.find('.navigation-area__account-title__button').trigger('click');

        expect(wrapper.find('.navigation-area__resources-title__button').text()).toMatch('Show');
        expect(wrapper.find('.navigation-area__account-title__button').text()).toMatch('Show');

        expect(wrapper.findAll('.navigation-area__item-container').length).toBe(3);

        await wrapper.find('.navigation-area__resources-title__button').trigger('click');
        await wrapper.find('.navigation-area__account-title__button').trigger('click');

        expect(wrapper.findAll('.navigation-area__item-container').length).toBe(7);

        await wrapper.find('.navigation-area__resources-title').trigger('mouseleave');
        await wrapper.find('.navigation-area__account-title').trigger('mouseleave');

        expect(wrapper.findAll('.navigation-area__resources-title__button').length).toBe(0);
        expect(wrapper.findAll('.navigation-area__account-title__button').length).toBe(0);

        expect(wrapper.findAll('.navigation-area__item-container').length).toBe(7);
    });
});
