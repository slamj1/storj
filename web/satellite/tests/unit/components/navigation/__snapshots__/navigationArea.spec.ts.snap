// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationArea snapshot not changed with project 1`] = `
<div class="navigation-area">
  <div class="navigation-area__wrap">
    <router-link-stub to="/project-dashboard" tag="a" event="click" class="navigation-area__logo">
      <logoicon-stub class="navigation-area__logo__img"></logoicon-stub>
      <logotexticon-stub class="navigation-area__logo__text"></logotexticon-stub>
    </router-link-stub>
    <projectselectionarea-stub class="project-selection-area"></projectselectionarea-stub>
    <p class="navigation-area__project-title">PROJECT</p>
    <router-link-stub to="/project-dashboard" tag="a" event="click" aria-label="Dashboard" class="navigation-area__item-container">
      <div class="navigation-area__item-container__link-container">
        <div><svg class="svg" width="19" height="17" viewBox="0 0 19 17" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path class="navigation-svg-path" fill-rule="evenodd" clip-rule="evenodd" d="M15.6691 16.8889H9.5H3.3309C1.29229 15.1294 0 12.5137 0 9.59331C0 4.29507 4.2533 0 9.5 0C14.7467 0 19 4.29507 19 9.59331C19 12.5137 17.7077 15.1294 15.6691 16.8889ZM12.9227 5.32961L14.4156 6.83704L9.93721 11.3594L8.44444 9.85192L12.9227 5.32961Z" fill="#7C8794"></path>
          </svg>
        </div>
        <h1 class="navigation-area__item-container__link-container__title">Dashboard</h1>
      </div>
    </router-link-stub>
    <router-link-stub to="/project-members" tag="a" event="click" aria-label="Team" class="navigation-area__item-container">
      <div class="navigation-area__item-container__link-container">
        <div><svg class="svg" width="21" height="18" viewBox="0 0 21 18" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path class="navigation-svg-path" opacity="0.8" fill-rule="evenodd" clip-rule="evenodd" d="M14.8165 0C17.7839 0 20.1895 2.40557 20.1895 5.373C20.1895 7.40389 19.0627 9.17161 17.4002 10.0851C19.4392 11.0968 20.8407 13.2 20.8407 15.6305V17.91H8.46655V15.6305C8.46655 13.13 9.94989 10.976 12.0846 10.0004C10.5035 9.06519 9.44345 7.34289 9.44345 5.373C9.44345 2.40557 11.849 0 14.8165 0ZM6.34991 0C7.36246 0 8.3096 0.280088 9.11806 0.767013C8.09966 2.02549 7.48964 3.62801 7.48964 5.373L7.49091 5.51065C7.51986 7.06803 8.03908 8.53529 8.92906 9.73552L8.97519 9.79612L8.94777 9.82267C7.42471 11.3194 6.51273 13.3922 6.51273 15.6305V17.91H0V15.6305C0 13.13 1.48335 10.976 3.61809 10.0004C2.037 9.06519 0.976909 7.34289 0.976909 5.373C0.976909 2.40557 3.38248 0 6.34991 0Z" fill="#909BA8"></path>
          </svg>
        </div>
        <h1 class="navigation-area__item-container__link-container__title">Team</h1>
      </div>
    </router-link-stub>
    <router-link-stub to="/api-keys" tag="a" event="click" aria-label="API Keys" class="navigation-area__item-container">
      <div class="navigation-area__item-container__link-container">
        <div><svg class="svg" width="19" height="19" viewBox="0 0 19 19" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path class="navigation-svg-path" opacity="0.8" fill-rule="evenodd" clip-rule="evenodd" d="M19 0L18.8777 3.30167L17.3804 3.33154L17.3114 4.86798L15.548 5.164L15.5552 6.62419L14.0737 6.63827L13.8958 8.28354L12.3349 9.84422L13.2397 13.2214C13.3996 13.8182 13.229 14.4549 12.7921 14.8918L9.19044 18.4934C8.75356 18.9303 8.1168 19.1009 7.52001 18.941L2.60009 17.6227C2.00331 17.4628 1.53716 16.9967 1.37725 16.3999L0.0589658 11.48C-0.100943 10.8832 0.0696777 10.2464 0.506556 9.80956L4.10819 6.20793C4.54507 5.77105 5.18183 5.60043 5.77862 5.76034L9.15552 6.66484L15.6983 0.122284L19 0ZM6.52703 12.473C5.78414 11.7301 4.57967 11.7301 3.83678 12.473C3.09389 13.2159 3.09389 14.4203 3.83678 15.1632C4.57967 15.9061 5.78414 15.9061 6.52703 15.1632C7.26992 14.4203 7.26992 13.2159 6.52703 12.473Z" fill="#909BA8"></path>
          </svg>
        </div>
        <h1 class="navigation-area__item-container__link-container__title">API Keys</h1>
      </div>
    </router-link-stub>
    <div class="navigation-area__resources-title custom">
      <p class="navigation-area__resources-title__title">RESOURCES</p>
      <!---->
      <!---->
    </div> <a href="https://documentation.storj.io" target="_blank" class="navigation-area__item-container">
      <div class="navigation-area__item-container__link-container">
        <docsicon-stub class="svg"></docsicon-stub>
        <h1 class="navigation-area__item-container__link-container__title docs-title">Docs</h1>
      </div>
    </a> <a href="mailto:support@storj.io" target="_blank" class="navigation-area__item-container support-item">
      <div class="navigation-area__item-container__link-container">
        <supporticon-stub class="svg"></supporticon-stub>
        <h1 class="navigation-area__item-container__link-container__title">Support</h1>
      </div>
    </a>
    <div class="navigation-area__account-title custom">
      <p class="navigation-area__account-title__title">ACCOUNT</p>
      <!---->
      <!---->
    </div>
    <div class="navigation-area__account-area">
      <router-link-stub to="/account/settings" tag="a" event="click" class="navigation-area__item-container account-item">
        <div class="navigation-area__item-container__link-container">
          <h1 class="navigation-area__item-container__link-container__title account-item">Settings</h1>
        </div>
      </router-link-stub>
      <router-link-stub to="/account/billing" tag="a" event="click" class="navigation-area__item-container account-item">
        <div class="navigation-area__item-container__link-container">
          <h1 class="navigation-area__item-container__link-container__title account-item">Billing</h1>
        </div>
      </router-link-stub>
    </div>
    <div class="divider custom-divider"></div>
  </div>
</div>
`;

exports[`NavigationArea snapshot not changed without project 1`] = `
<div class="navigation-area">
  <div class="navigation-area__wrap">
    <router-link-stub to="/onboarding-tour" tag="a" event="click" class="navigation-area__logo">
      <logoicon-stub class="navigation-area__logo__img"></logoicon-stub>
      <logotexticon-stub class="navigation-area__logo__text"></logotexticon-stub>
    </router-link-stub>
    <projectselectionarea-stub class="project-selection-area"></projectselectionarea-stub>
    <p class="navigation-area__project-title">PROJECT</p>
    <router-link-stub to="/project-dashboard" tag="a" event="click" aria-label="Dashboard" class="navigation-area__item-container disabled">
      <div class="navigation-area__item-container__link-container">
        <div><svg class="svg" width="19" height="17" viewBox="0 0 19 17" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path class="navigation-svg-path" fill-rule="evenodd" clip-rule="evenodd" d="M15.6691 16.8889H9.5H3.3309C1.29229 15.1294 0 12.5137 0 9.59331C0 4.29507 4.2533 0 9.5 0C14.7467 0 19 4.29507 19 9.59331C19 12.5137 17.7077 15.1294 15.6691 16.8889ZM12.9227 5.32961L14.4156 6.83704L9.93721 11.3594L8.44444 9.85192L12.9227 5.32961Z" fill="#7C8794"></path>
          </svg>
        </div>
        <h1 class="navigation-area__item-container__link-container__title">Dashboard</h1>
      </div>
    </router-link-stub>
    <router-link-stub to="/project-members" tag="a" event="click" aria-label="Team" class="navigation-area__item-container disabled">
      <div class="navigation-area__item-container__link-container">
        <div><svg class="svg" width="21" height="18" viewBox="0 0 21 18" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path class="navigation-svg-path" opacity="0.8" fill-rule="evenodd" clip-rule="evenodd" d="M14.8165 0C17.7839 0 20.1895 2.40557 20.1895 5.373C20.1895 7.40389 19.0627 9.17161 17.4002 10.0851C19.4392 11.0968 20.8407 13.2 20.8407 15.6305V17.91H8.46655V15.6305C8.46655 13.13 9.94989 10.976 12.0846 10.0004C10.5035 9.06519 9.44345 7.34289 9.44345 5.373C9.44345 2.40557 11.849 0 14.8165 0ZM6.34991 0C7.36246 0 8.3096 0.280088 9.11806 0.767013C8.09966 2.02549 7.48964 3.62801 7.48964 5.373L7.49091 5.51065C7.51986 7.06803 8.03908 8.53529 8.92906 9.73552L8.97519 9.79612L8.94777 9.82267C7.42471 11.3194 6.51273 13.3922 6.51273 15.6305V17.91H0V15.6305C0 13.13 1.48335 10.976 3.61809 10.0004C2.037 9.06519 0.976909 7.34289 0.976909 5.373C0.976909 2.40557 3.38248 0 6.34991 0Z" fill="#909BA8"></path>
          </svg>
        </div>
        <h1 class="navigation-area__item-container__link-container__title">Team</h1>
      </div>
    </router-link-stub>
    <router-link-stub to="/api-keys" tag="a" event="click" aria-label="API Keys" class="navigation-area__item-container disabled">
      <div class="navigation-area__item-container__link-container">
        <div><svg class="svg" width="19" height="19" viewBox="0 0 19 19" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path class="navigation-svg-path" opacity="0.8" fill-rule="evenodd" clip-rule="evenodd" d="M19 0L18.8777 3.30167L17.3804 3.33154L17.3114 4.86798L15.548 5.164L15.5552 6.62419L14.0737 6.63827L13.8958 8.28354L12.3349 9.84422L13.2397 13.2214C13.3996 13.8182 13.229 14.4549 12.7921 14.8918L9.19044 18.4934C8.75356 18.9303 8.1168 19.1009 7.52001 18.941L2.60009 17.6227C2.00331 17.4628 1.53716 16.9967 1.37725 16.3999L0.0589658 11.48C-0.100943 10.8832 0.0696777 10.2464 0.506556 9.80956L4.10819 6.20793C4.54507 5.77105 5.18183 5.60043 5.77862 5.76034L9.15552 6.66484L15.6983 0.122284L19 0ZM6.52703 12.473C5.78414 11.7301 4.57967 11.7301 3.83678 12.473C3.09389 13.2159 3.09389 14.4203 3.83678 15.1632C4.57967 15.9061 5.78414 15.9061 6.52703 15.1632C7.26992 14.4203 7.26992 13.2159 6.52703 12.473Z" fill="#909BA8"></path>
          </svg>
        </div>
        <h1 class="navigation-area__item-container__link-container__title">API Keys</h1>
      </div>
    </router-link-stub>
    <div class="navigation-area__resources-title custom">
      <p class="navigation-area__resources-title__title">RESOURCES</p>
      <!---->
      <!---->
    </div> <a href="https://documentation.storj.io" target="_blank" class="navigation-area__item-container disabled">
      <div class="navigation-area__item-container__link-container">
        <docsicon-stub class="svg"></docsicon-stub>
        <h1 class="navigation-area__item-container__link-container__title docs-title">Docs</h1>
      </div>
    </a> <a href="mailto:support@storj.io" target="_blank" class="navigation-area__item-container support-item disabled">
      <div class="navigation-area__item-container__link-container">
        <supporticon-stub class="svg"></supporticon-stub>
        <h1 class="navigation-area__item-container__link-container__title">Support</h1>
      </div>
    </a>
    <div class="navigation-area__account-title custom">
      <p class="navigation-area__account-title__title">ACCOUNT</p>
      <!---->
      <!---->
    </div>
    <div class="navigation-area__account-area">
      <router-link-stub to="/account/settings" tag="a" event="click" class="navigation-area__item-container account-item disabled">
        <div class="navigation-area__item-container__link-container">
          <h1 class="navigation-area__item-container__link-container__title account-item">Settings</h1>
        </div>
      </router-link-stub>
      <router-link-stub to="/account/billing" tag="a" event="click" class="navigation-area__item-container account-item disabled">
        <div class="navigation-area__item-container__link-container">
          <h1 class="navigation-area__item-container__link-container__title account-item">Billing</h1>
        </div>
      </router-link-stub>
    </div>
    <div class="divider custom-divider"></div>
  </div>
</div>
`;
